File: Transformers/TotalTimeTransformer.m
//
//  TotalTimeTransformer.m
//  Cog
//
//  Created by Christopher Snowhill on 6/15/22.
//

#import "TotalTimeTransformer.h"

@implementation TotalTimeTransformer

+ (Class)transformedValueClass {
	return [NSString class];
}
+ (BOOL)allowsReverseTransformation {
	return NO;
}

- (id)transformedValue:(id)value {
	if(value == nil) return @"";

	if([value isKindOfClass:[NSString class]]) {
		NSString *string = (NSString *)value;
		if([string length] > 0) {
			return [@"Total Duration: " stringByAppendingString:string];
		}
	}

	return @"";
}

@end

File: Transformers/TotalTimeTransformer.swift
import Foundation

@objc(TotalTimeTransformer)
class TotalTimeTransformer: ValueTransformer {
    override class func transformedValueClass() -> AnyClass {
        return NSString.self
    }

    override class func allowsReverseTransformation() -> Bool {
        return false
    }

    override func transformedValue(_ value: Any?) -> Any? {
        guard let value = value else {
            return ""
        }

        if let string = value as? String {
            if !string.isEmpty {
                return String(format: NSLocalizedString("Total duration: %@", comment: "Total duration for status"), string)
            }
        }

        return ""
    }
}
